### DISCUSSION TITLE
AWS S3 Storage support

### DISCUSSION THREAD

**Discussion by lohanidamodar** [appwrite/appwrite#852] [Ideas]
We been working an RFC for AWS S3 storage support as a next step to support external storage in Appwrite. You're all welcome to comment with your feedback.

https://github.com/appwrite/rfc/blob/main/009-s3-storage-support.md

**Comment by kodumbeats** [comment]
I strongly believe this feature should be provider-agnostic right out of the gate, instead of relegated to "future possibilities". Most cloud providers offer S3-compatible storage nowadays, and I [self-host](https://github.com/minio/minio) my own for personal use.

**Reply by eldadfux** [reply]
Might be a good Idea to offer the possibility to change the endpoint for allowing custom S3 integrations. Anyway we do plan to offer out of the box extensions to popular S3 providers like DigitalOcean which will just inherit the original implementation. I think this will make it easier to understand we do support a wide variety of providers and create a bit smaller friction when integrating.

This obviously also opens the door to a lot non-S3 providers like GCP, Azure, FTP (my favorite retro provider 😁) and more.

**Reply by MatricomHQ** [reply]
The lack of control for specifying the endpoint manually seems absurd considering S3 is a widely adapted protocol. For instance, iDrive is one of the more affordable S3 providers, and could have been easily configured with a mere switch of an endpoint. 

Is custom endpoints still on the roadmap? We can clone git and run without docker in the meantime, since its vital that we can access our S3 framework.

**Comment by dasheck0** [comment]
Not sure if this helps, but you could mount an s3 bucket on an EC2 instance and then host your appwrite instance there. The magic tool here is [s3fs-fuse](https://github.com/s3fs-fuse/s3fs-fuse). Not sure how this will react, when you scale you EC2 instances with a load balancer. But for a single instance, this works just fine.

**Comment by IwateKyle** [comment]
I agree that the s3 interface is ubiquitous so support for that would be an immediate win for newcomers.  
And, then, possibly a generic hook for custom code to fetch whatever it needs.
